modules/messages/server/services/message-to-influx.server.service.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+/**
+ * Module dependencies.
+ */
+var co = require('co');
+
+/**
+ * this module gets some statistical information for message from database
+ * and sends the data to influxService to add to influxdb
+ * @param {Object} message - a message returned from database (created or read)
+ * @param {Object} User - mongoose Model object (dependency)
+ * @param {Object} Message - mongoose Model object (dependency)
+ * @param {Object} influxService - a module with method writePoint() writes
+ * point to influxdb (dependency)
+ * @returns {Promise} - promise to be resolved with influxdb response or
+ * rejected with an error
+ */
+module.exports = function (message, { Message: Message, influxService: influxService }) {
+
+  return co(function * () {
+    // some variables used later, filled in inner scopes
+    let isFirstMessage,
+        isFirstReply,
+        replyTime,
+        msgLen;
+    let firstMessage,
+        firstReply;
+
+    // fixing some strange filling of message data
+    // (userFrom is not id but user object)
+    let userFrom = message.userFrom._id
+      ? message.userFrom._id
+      : message.userFrom;
+    let userTo = message.userTo._id
+      ? message.userTo._id
+      : message.userTo;
```

Comment:
Some background for this:
- in Messages table/Schema `userFrom` and `userTo` are Mongo-ids.
- When fetching them from the DB, they're populated against Users Schema and thus they are user objects. It's a Mongo thing — it knows what to do regardless if you pass it `keyId: 123` or `keyId: { _id: 123 }`.
- If user doesn't exist anymore when populating, you get back id instead of user object. Thus inconsistancy here.